compiler/src/dotty/tools/debug/ExtractExpression.scala

```diff
@@ -0,0 +1,322 @@
+
+  private def setField(tree: Tree)(qualifier: Tree, field: TermSymbol, rhs: Tree)(using Context): Tree =
+    val strategy = ReflectEvalStrategy.FieldAssign(field)
+    reflectEval(tree)(qualifier, strategy, List(rhs), tree.tpe)
+
+  private def callMethod(tree: Tree)(qualifier: Tree, method: TermSymbol, args: List[Tree])(using Context): Tree =
+    val strategy = ReflectEvalStrategy.MethodCall(method)
+    reflectEval(tree)(qualifier, strategy, args, tree.tpe)
+
+  private def callConstructor(tree: Tree)(qualifier: Tree, ctr: TermSymbol, args: List[Tree])(using Context): Tree =
+    val strategy = ReflectEvalStrategy.ConstructorCall(ctr, ctr.owner.asClass)
+    reflectEval(tree)(qualifier, strategy, args, tree.tpe)
+
+  private def reflectEval(tree: Tree)(
+      qualifier: Tree,
+      strategy: ReflectEvalStrategy,
+      args: List[Tree],
+      tpe: Type
+  )(using Context): Tree =
+    val evalArgs = List(
+      qualifier,
+      Literal(Constant(strategy.toString)),
+      JavaSeqLiteral(args, TypeTree(ctx.definitions.ObjectType))
+    )
+    cpy
+      .Apply(tree)(Select(This(config.expressionClass), termName("reflectEval")), evalArgs)
+      .withAttachment(ReflectEvalStrategy, strategy)
+
+  private def isStaticObject(symbol: Symbol)(using Context): Boolean =
+    symbol.is(Module) && symbol.isStatic && !symbol.is(JavaDefined) && !symbol.isRoot
+
+  private def isNonStaticObject(symbol: Symbol)(using Context): Boolean =
+    symbol.is(Module) && !symbol.isStatic && !symbol.isRoot
+
+  private def isLocalVariable(symbol: Symbol)(using Context): Boolean =
+    !symbol.is(Method) && symbol.isLocalToBlock
+
+  // Check if a term is accessible from the expression class
+  private def isAccessibleMember(tree: Tree)(using Context): Boolean =
+    val symbol = tree.symbol
+    symbol.owner.isType &&
+    !symbol.isPrivate &&
+    !symbol.is(Protected) &&
+    isTypeAccessible(widenDealiasQualifierType(tree))
+
+  private def widenDealiasQualifierType(tree: Tree)(using Context): Type =
+    tree match
+      case Ident(_) => tree.symbol.enclosingClass.thisType.widenDealias
+      case Select(qualifier, _) => qualifier.tpe.widenDealias
+      case Apply(fun, _) => widenDealiasQualifierType(fun)
+      case TypeApply(fun, _) => widenDealiasQualifierType(fun)
+
+  // Check if a type is accessible from the expression class
+  private def isTypeAccessible(tpe: Type)(using Context): Boolean =
+    def isPublic(sym: Symbol): Boolean =
+      !sym.isLocal && (sym.isPublic || sym.privateWithin.is(PackageClass))
+    tpe.forallParts {
+      case tpe: NamedType if tpe.symbol != NoSymbol =>
+        isLocalToExpression(tpe.symbol) || isPublic(tpe.symbol)
+      case _ => true
+    }
+
+  private def isLocalToExpression(symbol: Symbol)(using Context): Boolean =
+    val evaluateMethod = config.evaluateMethod
+    symbol.ownersIterator.exists(_ == evaluateMethod)
+
+private object ExtractExpression:
+  val name: String = "extract-expression"
```

Comment:
Phase names are usually camelCase:
```suggestion
  val name: String = "extractExpression"
```

Reply:
done